doc/manual/manual.md

```diff
@@ -3797,3 +3797,202 @@ N.B: The following commands must first be enabled using `enable_experimental`.
 
   Note that `Term`s derived from HDL modules are "first class", and are not restricted to `yosys_verify`: they may also be used with SAW's typical `Term` infrastructure like `sat`, `prove_print`, term rewriting, etc.
   `yosys_verify` simply provides a convenient and familiar interface, similar to `llvm_verify` or `jvm_verify`.
+
+# Bisimulation Prover
+
+SAW contains a bisimulation prover to prove that two terms simulate each other.
+This prover allows users to prove that two terms executing in lockstep satisfy
+some relation over the state of each circuit and their outputs.  This type of
+proof is useful in demonstrating the eventual equivalence of two circuits, or
+of a circuit and a functional specification.  SAW enables these proofs with the
+experimental `prove_bisim` command:
+
+~~~~
+prove_bisim : ProofScript () -> Term -> Term -> Term -> TopLevel ProofResult
+~~~~
+
+When invoking `prove_bisim strat relation lhs rhs`, the arguments represent the
+following:
+
+1. `strat`: A proof strategy to use during verification.
+2. `relation`: A relation between `lhs` and `rhs`.  This relation must have the
+   type `(lhsState, output) -> (rhsState, output) -> Bit`. The relation's first
+   argument is a pair consisting of `lhs`'s state and output following
+   execution. The relation's second argument is a pair consisting of `rhs`'s
+   state and output following execution. `relation` then then returns a `Bit`
+   indicating whether the two arguments satisfy the bisimulation relation. That
+   is, whether the terms simulate each other.
+3. `lhs`: A term that simulates `rhs`. `lhs` must have the type
+   `(lhsState, input) -> (lhsState, output)`.  The first argument to `lhs` is a
+   tuple containing the internal state of `lhs`, as well as the input to `lhs`.
+   `lhs` returns a tuple containing its internal state after execution, as well
+   as its output.
+4. `rhs`: A term that simulates `lhs`. `rhs` must have the type
+   `(rhsState, input) -> (rhsState, output)`.  The first argument to `rhs` is a
+   tuple containing the internal state of `rhs`, as well as the input to `rhs`.
+   `rhs` returns a tuple containing its internal state after execution, as well
+   as its output.
+
+`prove_bisim` returns a `ProofResult` indicating whether `lhs` and `rhs` are
+bisimilar, given the relation `relation`.
```

Comment:
Is it worth stating `forall s1 s2 in out1 out2. relation (s1, out1) (s2, out2) -> relation (lhs (s1, in)) (rhs (s2, in))` somewhere in here?

Reply:
I added a short subsection about this at the end